### ISSUE TITLE
Unique indices not being created on relationship fields

### ISSUE THREAD

**Issue by dodiameer** [payloadcms/payload#859]
# Bug Report

I'm attempting to create a 1-1 relationship between a `PublicUsers` collection and `Carts` collection, where the relationship field is present on the `Carts` collection. While an index is being created, this index doesn't have the `unique` present.

## Expected Behavior

Using `unique: true` on a relationship field, the created index should have the unique property present

## Current Behavior

An index is created without the unique property present

## Possible Solution

_No answer_

## Steps to Reproduce
1. Create new payload app
2. Create a collection
3. Create another collection with a relationship field with `relationTo: 'first-collection'` and `unique: true`
4. Create an entry in the first collection
5. Create two entries referencing the same entry in the second collection

My example has a `PublicUsers` and `Carts` collection that look like the following:

```ts
// PublicUsers.ts
import { CollectionConfig } from "payload/types";

export const PublicUsers: CollectionConfig = {
  slug: "public-users",
  auth: {
    verify: true,
  },
  admin: {
    useAsTitle: "email",
  },
  access: {
    // Irrelevant
  },
  hooks: {
    // Irrelevant
  },
  fields: [
    {
      name: "firstName",
      type: "text",
      maxLength: 50,
      required: true,
    },
    // The rest of the fields
  ],
};
```

```ts
// Carts.ts
import { CollectionConfig } from "payload/types";
import { Products } from "./Products";

export const Carts: CollectionConfig = {
  slug: "carts",
  fields: [
	// This is the field with the problem
    {
      name: "user",
      type: "relationship",
      relationTo: "public-users",
      hasMany: false,
      unique: true,
      required: true,
    },
  ],
};
```

The index created doesn't have the unique property present.

![image](https://user-images.githubusercontent.com/61318540/182159227-faac3579-597a-497b-866b-d6df6c42a1a9.png)

## Detailed Description

- Payload version: 1.0.11
- Node version: 16.14.2
- MongoDB version: 5.0.9

**Comment by DanRibbens** [comment]
Hey @dodiameer,

Is it possible that your relationship field had `index: true` and you later change it to unique? Payload won't change existing indexes in the database and I'm guessing that is what has happened in your case. 

If an index in MongoDB ever diverges from the Payload config, the quickest fix is to manually delete the index and start Payload again or create it manually outside of Payload without stopping and starting payload if you're running in production.

Can you try and let me know if the unique index still isn't right?

**Comment by DanRibbens** [comment]
Fixed in v1.0.12

Let us know if you still have any trouble!
